### PR TITLE
Implement prune only stagnant check mode

### PR DESCRIPTION
**Pull request by vstakhov**
As discussed, we are not sure about interaction between finality lag and stagnant checks. This PR tries to introduce a trade-off where stagnant entries are just cleaned after some long period of time (25 hours) without the actual checks of stagnant blocks. This allows to keep database clean from stagnant entries that are otherwise kept indefinitely when stagnant checks are disabled. On the other hand, this mode does not interfere with finality and has quite a large margin to prevent active entries to be removed.

### CODE REVIEW

File: node/service/src/lib.rs [hunk 1]
```diff
@@ -1477,6 +1478,7 @@ fn revert_chain_selection(db: Arc<dyn Database>, hash: Hash) -> sp_blockchain::R
 	let config = chain_selection_subsystem::Config {
 		col_data: parachains_db::REAL_COLUMNS.col_chain_selection_data,
 		stagnant_check_interval: chain_selection_subsystem::StagnantCheckInterval::never(),
+		stagnant_check_mode: chain_selection_subsystem::StagnantCheckMode::PruneOnly,
```
Comment (ordian) [inline node/service/src/lib.rs:1481]: will pruning be ever called if check interval is `never`?

Comment (vstakhov) [reply node/service/src/lib.rs:1481]: No, it won't. But as far as I understand, this code is called merely for reversion of the chain selection and notihing else. Please correct me if I'm wrong.

Comment (ordian) [reply node/service/src/lib.rs:1481]: Oh, sorry, I missed some part of the diff.

File: node/core/chain-selection/src/tree.rs [hunk 2]
```diff
@@ -594,6 +594,41 @@ pub(super) fn detect_stagnant<'a, B: 'a + Backend>(
...
+	let mut backend = OverlayedBackend::new(backend);
+
+	let (min_ts, max_ts) = match stagnant_up_to.len() {
+		0 => (0 as Timestamp, 0 as Timestamp),
+		1 => (stagnant_up_to[0].0, stagnant_up_to[0].0),
+		n => (stagnant_up_to[0].0, stagnant_up_to[n - 1].0),
+	};
+
+	// As this is in ascending order, only the earliest stagnant
+	// blocks will involve heavy viability propagations.
```
Comment (ordian) [inline node/core/chain-selection/src/tree.rs:615]: we do not propagate viability here
```suggestion
```

**Review by ordian — APPROVED** [review]

**Review by rphmeier — APPROVED** [review]

**Comment by rphmeier** [comment]
bot merge

**Pull request merged**